Add the workshop report. (#10)
rats-endorsements-distribution/linaro-connect-25: workshop-report.md

# Endorsement API Workshop Report

## Quick Recap

> The meeting focused on endorsement and reference value distribution for attestation, covering topics such as creation protocols, existing distribution services, and standardization efforts.

Reviewer: Minor grammar nit:

```suggestion
The meeting focused on endorsement and reference value distribution for attestation, covering topics such as creating protocols, existing distribution services, and standardization efforts.
```

## Meeting Summary

### Confidential Computing Endorsement Distribution Challenges
The meeting begins with introductions from participants, who share their backgrounds and interests in endorsement distribution for attestation.
Paul Howard from Arm introduces the topic, emphasizing the need to address fragmentation risks in endorsement distribution across hardware, software, and firmware.
Attendees include representatives from Linaro, Microsoft, Alibaba Cloud, Huawei, Fujitsu, and other organizations involved in confidential computing and remote attestation.
The group discusses the agenda, which includes talks on endorsement creation protocols, existing distribution services, interaction models and APIs, and standardization efforts.
Michael Richardson is set to present first on the endorsement creation protocol and proof of possession used by auditors.

### Proof of Presence Endorsement Concept
Michael Richardson presents a concept for a "Proof of Presence Endorsement" to verify the physical location of network devices.
This endorsement is created by an auditor who physically inspects the device and exchanges signed data with it through a console protocol.

> The process involves the auditor connecting to the device, verifying its identity, and creating a signed attestation about its location and connections.

Reviewer 2: ```suggestion
The process involves the auditor connecting to the device, verifying its identity, and creating a signed endorsement about its location and connections.
```

Reviewer: I agree with @mcr's proposal here.

### Versioning and Timeliness in Endorsements
The discussion focuses on versioning and timeliness of endorsements in attestation systems.
Michael and Greg emphasize the importance of versioning for all components, including topologies and shapes.
They agree that timeliness signals are critical for endorsements, but caution against specifying implementation details like revocation mechanisms in the requirements.
The group discusses strategies for handling endorsement expiration and validity periods, with Thomas suggesting that attestation results should not be limited by the shortest endorsement validity period.
They also touch on the differences between recentness and freshness of endorsements. 

### Conclusions from the Requirements Conversation
- Shefali: endorsements need timeliness/freshness signals (incl. explicit revocation) for statements previously made by the endorser
- Greg: the parties in the flow have a voice to say how fresh the endorsement is (Supply chain vs Repackager — CSP)
- Jag: attester shapes change too, not just the measured component
- Michael: Versions: we need to assign versions to things

> Synth: attester shapes are endorsements and must have versions

Reviewer: Should this be part of the bullet list like the entries above and below it?

```suggestion
- Synth: attester shapes are endorsements and must have versions
```

Paul walks through a simple, self-contained command-line program that makes a single API call to the RIM service to obtain a CoRIM, and then shows how its inner CoMID and reference value structures could be copied across to simulate a CoSERV query/response transaction.
Thomas points out that he has developed a plug-in framework as part of his PoC efforts in Veraison, and it would be possible to take some of the same code into that framework, allowing Veraison to act as a CoSERV proxy and aggregator for NVIDIA RIMs.
This could become part of a future hackathon, possibly at the forthcoming IETF meeting in Madrid.
Ned asks about how the signatures were being handled, and Paul responds that there is currently no signature handling or re-signing taking place, because this was only a very brief and constrained prototyping exercise.

### CoSERV API and Implementation
The group discusses the CoSERV API and its capability to return signed CoRIM from the original source.
Thomas clarifies that while this functionality is not yet implemented, it is a core requirement to optionally bundle the original data for transparency.

> The conversation then shifts to the timeline for implementing CoSERV support in Verizon, with Thomas explaining that it's currently in a branch but not ready for mainline due to the evolving nature of the specification.

Reviewer 3: ```suggestion
The conversation then shifts to the timeline for implementing CoSERV support in Veraison, with Thomas explaining that it's currently in a branch but not ready for mainline due to the evolving nature of the specification.
```

### CDDL Data Model for In-ToTo Attestations
Henk reports on his morning hackathon activities.
He provides an overview of In-Toto attestations, explaining their potential use in showing the provenance of reference values in the CoRIM space.
He shares his exploration of the In-Toto attestation specification, noting its similarities to their existing system and the potential for mapping it to their schema, though some challenges remain due to the complexity of certain predicates.

### Actions and Next Steps
- Michael to further develop the endorsement creation protocol for physical audits of device locations.
- Thomas and Michael to determine which working group the endorsement creation protocol document should be submitted to.

> - Kathleen to draft a liaison request for Etsy and ISO documents related to confidential computing standards.

Reviewer 3: ```suggestion
- Kathleen to draft a liaison request for ETSI and ISO documents related to confidential computing standards.
```

- Nicolae to send Kathleen the contact information for the head of the ISO work group on confidential computing.

> - Thomas to provide Kathleen with specific document requests for Etsy and ISO.

Reviewer 3: Not sure about this.  I think it’s s/Thomas/Nicolae/

Also s/Etsy/ETSI/ ;-)

- Confidential computing community to consider ways to align work in ETSI with IETF use cases.
- Workshop participants to continue discussion on CoSERV and its implementation as a standard for endorsement and reference value conveyance.
- Paul to add a report summarizing the meeting's discussions and outcomes to the project repository.
- Paul to continue working on the example proxy plugin for NVIDIA RIMs in the CoSERV implementation.

> - Paul and Thomas to consider adding support for returning original signed Co-RIMs as part of the Coserve API.

Reviewer 3: ```suggestion
- Paul and Thomas to consider adding support for returning original signed CoRIMs as part of the CoSERV API.
```

> - Paul and Thomas to develop deployment scenarios to clarify different use cases for Coserve.

Reviewer 3: ```suggestion
- Paul and Thomas to develop deployment scenarios to clarify different use cases for CoSERV.
```

> - Thomas to continue developing the CoSERV proof-of-concept in their project branch.

Reviewer 4: Could we add a link to the project branch?

Author: Thank you, yes, done.
